### ISSUE TITLE
Code available for all examples in the book?

### ISSUE THREAD

**Issue by ananovo73** [PacktPublishing/Enhancing-Deep-Learning-with-Bayesian-Inference#11]
I am happy having found this really interesting book. Unfortunately, it is not clear to me where I can find the code examples for e.g. Chapter 2, 3 etc.

**Comment by gietema** [comment]
Hi, thank you! Sorry for the late reply, we'll be adding code for all chapters this month.

**Comment by HolyGold-DUT** [comment]
I'm happy to have found this book, but I'm not able to reproduce the code in chapter 2 by following the environment configuration in the book. Here is my environment configuration
# Name                    Version                   Build  Channel
abseil-cpp                20210324.2           h0e60522_0    conda-forge
absl-py                   2.1.0              pyhd8ed1ab_0    conda-forge
aiohappyeyeballs          2.4.4              pyhd8ed1ab_0    conda-forge
aiohttp                   3.11.9           py39hf73967f_0    conda-forge
aiosignal                 1.3.1              pyhd8ed1ab_0    conda-forge
arviz                     0.17.1           py39haa95532_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
astor                     0.8.1              pyh9f0ad1d_0    conda-forge
astunparse                1.6.3              pyhd8ed1ab_2    conda-forge
async-timeout             5.0.1              pyhd8ed1ab_1    conda-forge
attrs                     24.2.0             pyh71513ae_0    conda-forge
blinker                   1.9.0              pyhff2d567_0    conda-forge
brotli                    1.1.0                h2466b09_2    conda-forge
brotli-bin                1.1.0                h2466b09_2    conda-forge
brotli-python             1.1.0            py39ha51f57c_2    conda-forge
bzip2                     1.0.8                h2466b09_7    conda-forge
ca-certificates           2024.11.26           haa95532_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cachetools                4.2.4              pyhd8ed1ab_0    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.17.1           py39ha55e580_0    conda-forge
charset-normalizer        3.4.0              pyhd8ed1ab_1    conda-forge
click                     8.1.7           win_pyh7428d3b_1    conda-forge
cloudpickle               3.1.0              pyhd8ed1ab_1    conda-forge
colorama                  0.4.6              pyhd8ed1ab_1    conda-forge
contourpy                 1.3.0            py39h2b77a98_2    conda-forge
cryptography              39.0.0           py39h58e9bdb_0    conda-forge
cudatoolkit               11.3.1              hf2f0253_13    conda-forge
cudnn                     8.2.1.32             h754d62a_0    conda-forge
cxx-compiler              1.8.0                h91493d7_1    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_1    conda-forge
dill                      0.3.9              pyhd8ed1ab_1    conda-forge
dm-tree                   0.1.7            py39h2e25243_0    conda-forge
fastprogress              1.0.3              pyhd8ed1ab_0    conda-forge
filelock                  3.16.1             pyhd8ed1ab_1    conda-forge
flatbuffers               24.3.25                  pypi_0    pypi
fonttools                 4.55.1           py39hf73967f_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
frozenlist                1.5.0            py39ha55e580_0    conda-forge
gast                      0.6.0                    pypi_0    pypi
giflib                    5.2.2                h64bf75a_0    conda-forge
google-auth               1.35.0             pyh6c4a22f_0    conda-forge
google-auth-oauthlib      0.4.6              pyhd8ed1ab_0    conda-forge
google-pasta              0.2.0              pyhd8ed1ab_1    conda-forge
grpcio                    1.68.1                   pypi_0    pypi
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
h5netcdf                  1.4.1              pyhd8ed1ab_0    conda-forge
h5py                      3.12.1                   pypi_0    pypi
hdf5                      1.14.0          nompi_h97a5375_103    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       68.2                 h0e60522_0    conda-forge
idna                      3.10               pyhd8ed1ab_1    conda-forge
importlib-metadata        8.5.0              pyha770c72_1    conda-forge
importlib-resources       6.4.5              pyhd8ed1ab_1    conda-forge
importlib_resources       6.4.5              pyhd8ed1ab_1    conda-forge
intel-openmp              2024.2.1          h57928b3_1083    conda-forge
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   hcfcfb64_3    conda-forge
keras                     3.7.0                    pypi_0    pypi
keras-preprocessing       1.1.2              pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.7            py39h2b77a98_0    conda-forge
krb5                      1.20.1               h6609f42_0    conda-forge
lcms2                     2.15                 ha5c8aab_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libaec                    1.1.3                h63175ca_0    conda-forge
libblas                   3.9.0              25_win64_mkl    conda-forge
libbrotlicommon           1.1.0                h2466b09_2    conda-forge
libbrotlidec              1.1.0                h2466b09_2    conda-forge
libbrotlienc              1.1.0                h2466b09_2    conda-forge
libcblas                  3.9.0              25_win64_mkl    conda-forge
libclang                  18.1.1                   pypi_0    pypi
libcurl                   8.1.2                h68f0423_0    conda-forge
libdeflate                1.17                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libgpuarray               0.7.6             h8ffe710_1003    conda-forge
libhwloc                  2.11.2          default_hc8275d1_1000    conda-forge
libiconv                  1.17                 hcfcfb64_2    conda-forge
liblapack                 3.9.0              25_win64_mkl    conda-forge
libpng                    1.6.43               h19919ed_0    conda-forge
libprotobuf               3.14.0               h7755175_0    conda-forge
libsqlite                 3.47.0               h2466b09_1    conda-forge
libssh2                   1.10.0               h680486a_3    conda-forge
libtiff                   4.5.0                hf8721a0_2    conda-forge
libwebp-base              1.4.0                hcfcfb64_0    conda-forge
libxcb                    1.13                          0    conda-forge
libxml2                   2.13.5               h24da03e_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libzlib                   1.2.13               h2466b09_6    conda-forge
m2w64-binutils            2.25.1                        5    conda-forge
m2w64-bzip2               1.0.6                         6    conda-forge
m2w64-crt-git             5.0.0.4636.2595836               2    conda-forge
m2w64-gcc                 5.3.0                         6    conda-forge
m2w64-gcc-ada             5.3.0                         6    conda-forge
m2w64-gcc-fortran         5.3.0                         6    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gcc-objc            5.3.0                         6    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-headers-git         5.0.0.4636.c0ad18a               2    conda-forge
m2w64-isl                 0.16.1                        2    conda-forge
m2w64-libiconv            1.14                          6    conda-forge
m2w64-libmangle-git       5.0.0.4509.2e5a9a2               2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
m2w64-make                4.1.2351.a80a8b8               2    conda-forge
m2w64-mpc                 1.0.3                         3    conda-forge
m2w64-mpfr                3.1.4                         4    conda-forge
m2w64-pkg-config          0.29.1                        2    conda-forge
m2w64-toolchain           5.3.0                         7    conda-forge
m2w64-tools-git           5.0.0.4592.90b8472               2    conda-forge
m2w64-windows-default-manifest 6.4                           3    conda-forge
m2w64-winpthreads-git     5.0.0.4634.697f757               2    conda-forge
m2w64-zlib                1.2.8                        10    conda-forge
mako                      1.3.6              pyhd8ed1ab_1    conda-forge
markdown                  3.7                      pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2            py39hf73967f_1    conda-forge
matplotlib                3.9.1            py39hcbf5309_1    conda-forge
matplotlib-base           3.9.1            py39h5376392_2    conda-forge
mdurl                     0.1.2                    pypi_0    pypi
mkl                       2024.2.2            h66d3029_14    conda-forge
mkl-service               2.4.2            py39hdab2772_0    conda-forge
ml-dtypes                 0.4.1                    pypi_0    pypi
msys2-conda-epoch         20160418                      1    conda-forge
multidict                 6.1.0            py39hf73967f_1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
namex                     0.0.8                    pypi_0    pypi
numpy                     1.26.4           py39hddb5d58_0    conda-forge
oauthlib                  3.2.2              pyhd8ed1ab_0    conda-forge
openjpeg                  2.5.0                ha2aaf27_2    conda-forge
openssl                   1.1.1w               hcfcfb64_0    conda-forge
opt-einsum                3.4.0                    pypi_0    pypi
opt_einsum                3.3.0              pyhc1e730c_2    conda-forge
optree                    0.13.1                   pypi_0    pypi
packaging                 24.2               pyhd8ed1ab_2    conda-forge
pandas                    2.2.3            py39h2366fc2_1    conda-forge
patsy                     1.0.1              pyhff2d567_0    conda-forge
pillow                    9.4.0            py39hcebd2be_1    conda-forge
pip                       24.2             py39haa95532_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
propcache                 0.2.0            py39ha55e580_2    conda-forge
protobuf                  5.29.0                   pypi_0    pypi
pthreads-win32            2.9.1                h2466b09_4    conda-forge
pyasn1                    0.6.1              pyhd8ed1ab_2    conda-forge
pyasn1-modules            0.4.1              pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyh29332c3_1    conda-forge
pygments                  2.18.0                   pypi_0    pypi
pygpu                     0.7.6           py39h5d4886f_1003    conda-forge
pyjwt                     2.10.1             pyhd8ed1ab_0    conda-forge
pymc3                     3.11.2             pyh4f5629e_2    conda-forge
pyopenssl                 23.2.0             pyhd8ed1ab_1    conda-forge
pyparsing                 3.2.0              pyhd8ed1ab_2    conda-forge
pyqt                      5.12.3           py39hcbf5309_8    conda-forge
pyqt-impl                 5.12.3           py39h415ef7b_8    conda-forge
pyqt5-sip                 4.19.18          py39h415ef7b_8    conda-forge
pyqtchart                 5.12             py39h415ef7b_8    conda-forge
pyqtwebengine             5.12.1           py39h415ef7b_8    conda-forge
pysocks                   1.7.1              pyh09c184e_7    conda-forge
python                    3.9.15          h0269646_0_cpython    conda-forge
python-dateutil           2.9.0.post0        pyhff2d567_1    conda-forge
python-flatbuffers        1.12               pyhd8ed1ab_1    conda-forge
python-tzdata             2024.2             pyhd8ed1ab_1    conda-forge
python_abi                3.9                      5_cp39    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyu2f                     0.1.5              pyhd8ed1ab_0    conda-forge
qhull                     2020.2               hc790b64_5    conda-forge
qt                        5.12.9               h5909a2a_4    conda-forge
requests                  2.32.3             pyhd8ed1ab_1    conda-forge
requests-oauthlib         2.0.0              pyhd8ed1ab_0    conda-forge
rich                      13.9.4                   pypi_0    pypi
rsa                       4.9                pyhd8ed1ab_0    conda-forge
scikit-learn              1.5.2            py39hdd013cc_1    conda-forge
scipy                     1.13.1           py39h1a10956_0    conda-forge
seaborn                   0.13.2               hd8ed1ab_2    conda-forge
seaborn-base              0.13.2             pyhd8ed1ab_2    conda-forge
semver                    3.0.2              pyhd8ed1ab_0    conda-forge
setuptools                75.1.0           py39haa95532_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
six                       1.16.0             pyhd8ed1ab_1    conda-forge
snappy                    1.2.1                h23299a8_0    conda-forge
sqlite                    3.45.3               h2bbff1b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
statsmodels               0.14.4           py39h4b0a98a_0    conda-forge
tbb                       2021.13.0            h62715c5_1    conda-forge
tensorboard               2.18.0                   pypi_0    pypi
tensorboard-data-server   0.7.2                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1              pyhd8ed1ab_0    conda-forge
tensorflow                2.18.0                   pypi_0    pypi
tensorflow-base           2.5.0           gpu_py39hb3da07e_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
tensorflow-estimator      2.5.0              pyh81a9013_1    conda-forge
tensorflow-intel          2.18.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.31.0                   pypi_0    pypi
tensorflow-probability    0.13.0             pyhd8ed1ab_0    conda-forge
termcolor                 2.5.0              pyhd8ed1ab_0    conda-forge
theano                    1.0.5                    pypi_0    pypi
theano-pymc               1.1.2            py39h415ef7b_0    conda-forge
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
tornado                   6.4.2            py39ha55e580_0    conda-forge
typing-extensions         4.12.2               hd8ed1ab_1    conda-forge
typing_extensions         4.12.2             pyha770c72_1    conda-forge
tzdata                    2024b                h04d1e81_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
ucrt                      10.0.22621.0         h57928b3_1    conda-forge
unicodedata2              15.1.0           py39ha55e580_1    conda-forge
urllib3                   2.2.3              pyhd8ed1ab_1    conda-forge
vc                        14.40                h2eaa2aa_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
vc14_runtime              14.42.34433         he29a5d6_23    conda-forge
vs2015_runtime            14.42.34433         hdffcdeb_23    conda-forge
vs2017_win-64             19.16.27033         hddac466_20    conda-forge
vs2019_win-64             19.29.30139         he1865b1_23    conda-forge
vswhere                   3.1.7                h57928b3_0    conda-forge
werkzeug                  3.1.3              pyhd8ed1ab_1    conda-forge
wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
win_inet_pton             1.1.0              pyh7428d3b_8    conda-forge
wrapt                     1.17.0           py39ha55e580_0    conda-forge
xarray                    2024.7.0           pyhd8ed1ab_0    conda-forge
xarray-einstats           0.7.0              pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yarl                      1.18.0           py39ha55e580_0    conda-forge
zipp                      3.21.0             pyhd8ed1ab_1    conda-forge
zlib                      1.2.13               h2466b09_6    conda-forge
zstandard                 0.23.0           py39h9bf74da_1    conda-forge
zstd                      1.5.6                h0ea2cb4_0    conda-forge

**Comment by HolyGold-DUT** [comment]
> 我很高兴找到了这本非常有趣的书。不幸的是，我不清楚在哪里可以找到第 2 章、第 3 章等的代码示例。

你好我想问问你复现成功了么，ch02的代码

**Comment by gietema** [comment]
Chapter 2 to uses the pretty old `pymc3` library which is not compatible with modern numpy versions.
I'd recommend first installing `pymc3`, then `numpy` and other dependencies (`seaborn`, `pandas` and `matplotlib`).
